lib/GADS/Schema/Result/Site.pm

```diff
@@ -331,4 +337,46 @@ sub user_fields
     return @fields;
 }
 
+sub load_logo {
+  my $self = shift;
+  my $filecheck = GADS::Filecheck->instance();
+
+  my $logo_path = $self->create_temp_logo;
+
+  return undef unless $logo_path;
+
+  my $result = +{
+    'content_type'=> $filecheck->get_filetype($logo_path),
+    'filename' => 'logo',
+    'data' => $self->site_logo
+  };
+
+  return $result;
+}
+
+sub create_temp_logo {
+  my $self = shift;
+  my $logo = $self->site_logo;
+
+  return undef unless $logo;
+
+  my ($fh, $filename) = tempfile(UNLINK => 1);
+  print $fh $logo;
+  close $fh;
+
+  return $filename;
+}
+
+sub read_security_marking {
+  my $self = shift;
+  my $marking = $self->security_marking;
+
+  return $marking if $marking;
+
+  my $config = GADS::Config->instance;
+  $marking = $config && $config->gads && $config->gads->{header};
```

Comment:
See similar comments above, this section could probably be collapsed to a couple of lines.